packages/widget/src/components/address-input/address-input.ts

```diff
@@ -0,0 +1,90 @@
+import { LitElement, html } from 'lit';
+import type { HTMLTemplateResult } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { ethers } from 'ethers';
+import { Network } from '@buildwithsygma/sygma-sdk-core';
+import { when } from 'lit/directives/when.js';
+import { validateSubstrateAddress } from '../../utils';
+import { styles } from './styles';
+
+@customElement('sygma-address-input')
+export class AddressInput extends LitElement {
+  static styles = styles;
+  @property({
+    type: String
+  })
+  address: string = '';
+
+  @property({ attribute: false })
+  onAddressChange: (address: string) => void = () => {};
+
+  @property({
+    type: String
+  })
+  network: Network = Network.EVM;
+
+  @state()
+  errorMessage?: string;
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    if (this.address) {
+      this.handleAddressChange({
+        target: { value: this.address }
+      } as unknown as Event);
+    }
+  }
+
+  private handleAddressChange = ({ target }: Event): void => {
+    const { value } = target as HTMLInputElement;
+
+    if (this.errorMessage) {
+      this.errorMessage = undefined;
+    }
+
+    if (value !== '') {
```

Comment:
I agree, but the way I see it is that we want to display the error message either if the value is empty string or an invalid/valid address. Since this is not going to be undefined, I can't imagine how to do an early return over this. We don't want this to pass for empty string to the callback

Reply:
@sztok7 let me know if the pushed solution makes sense to you